src/pretix/static/pretixpresale/js/walletdetection.js

```diff
@@ -0,0 +1,120 @@
+        // providing a lot of information.
+        // So for the time being, we only check if Google Pay is available in TEST-mode, which should hopefully give us a
+        // good enough idea if Google Pay could be present on this device; even though there are still a lot of other
+        // factors that could inhibit Google Pay from actually being offered to the customer.
+
+        const baseRequest = {
+            apiVersion: 2,
+            apiVersionMinor: 0
+        };
+        const tokenizationSpecification = {
+            type: 'PAYMENT_GATEWAY',
+            parameters: {
+                'gateway': 'example',
+                'gatewayMerchantId': 'exampleGatewayMerchantId'
+            }
+        };
+        const allowedCardNetworks = ["AMEX", "DISCOVER", "INTERAC", "JCB", "MASTERCARD", "VISA"];
+        const allowedCardAuthMethods = ["PAN_ONLY", "CRYPTOGRAM_3DS"];
+        const baseCardPaymentMethod = {
+            type: 'CARD',
+            parameters: {
+                allowedAuthMethods: allowedCardAuthMethods,
+                allowedCardNetworks: allowedCardNetworks
+            }
+        };
+        const cardPaymentMethod = Object.assign(
+            {tokenizationSpecification: tokenizationSpecification},
+            baseCardPaymentMethod
+        );
+
+        return $.ajax({
+            url: 'https://pay.google.com/gp/p/js/pay.js',
+            dataType: 'script',
+            success: function () {
+                const paymentsClient = new google.payments.api.PaymentsClient({environment: 'TEST'});
+                const isReadyToPayRequest = Object.assign({}, baseRequest);
+                isReadyToPayRequest.allowedPaymentMethods = [baseCardPaymentMethod];
+
+                paymentsClient.isReadyToPay(isReadyToPayRequest)
+                    .then(function (response) {
+                        if (response.result) {
+                            return true;
+                        }
+                    })
+                    .catch(function (err) {
+                        return false;
+                    });
+            },
+        });
+    },
+    name_map: {
+        applepay: gettext('Apple Pay'),
+        googlepay: gettext('Google Pay'),
+    }
+}
+
+$(function () {
+    let requestedWallets = Array();
+    let paymentMethods = $('[data-wallets][data-wallets!=""]');
+
+    paymentMethods.each(function () {
+        let $s = $(this);
+        requestedWallets = requestedWallets.concat($s.data("wallets").split("|"));
+    })
+
+    // Filtering out any doubles
+    requestedWallets = new Set(requestedWallets);
+
+    // Perform the actual check *only* on the requested wallets, if they are supported by the browser
+    let availableWallets = Array();
+    requestedWallets.forEach(function (it) {
+        if (walletdetection[it]()) {
+            availableWallets.push(it);
+        }
+    });
```

Comment:
I have not tested this, but wouldn’t `walletdetection[it]()` with `it == 'googlepay'` always return true, as $.ajax without `async/await` returns a jqXHR-object?

With `async/await` (see comments above on async applepay/googlepay) I’d shorten this to:

```suggestion
    let paymentMethods = $('[data-wallets]');
    let availableWallets = paymentMethods
        .map(pm => pm.getAttribute("data-wallets").split("|"))
        .flat()
        .filter(function(item, pos, self) {
            return item && self.indexOf(item) == pos;
        })
        .filter(function(item) {
            return await walletdetection[item]();
        })
```